hastexo/hastexo.py

```diff
@@ -518,7 +519,14 @@ def get_stack_name(self):
         # Get the course id and anonymous user id, and derive the stack name
         # from them
         course_id, student_id = self.get_block_ids()
-        return "%s_%s_%s" % (course_id.course, course_id.run, student_id)
+        stack_name = "%s_%s_%s" % (course_id.course, course_id.run, student_id)
+
+        # make sure that the stack name doesn't include any unsafe characters
+        # (according to Nova, for creating a keypair)
+        safe_chars = "_- " + string.digits + string.ascii_letters
+        stack_name = "".join(x if x in safe_chars else '_' for x in stack_name)
```

Comment:
While Nova may allow whitespace and hyphens in keypair names, I'm not entirely sure if we want them in stack names.

I'd just go with this:

```python
# Replace anything in the stack name that is not an ASCII letter or digit with an underscore
replace_pattern = '[^%s%s]' % (string.digits, string.ascii_letters)
stack_name = re.sub(re.compile(replace_pattern), '_', stack_name)
```

What do you think?

Reply:
Sounds good,  I had my doubts about the whitespace as well.